File: javascript/prime-factors/prime-factors.js
function PrimeFactors() {}

PrimeFactors.prototype.for = function(num) {
  let testNum = num;
  let divisor = 2;
  const factors = [];

  while (testNum > 1) {
    if (testNum % divisor === 0) {
      testNum /= divisor;
      factors.push(divisor);
      // retry successful factor
      divisor -= 1;
    }
    divisor += 1;
  }
  return factors;
};

module.exports = PrimeFactors;

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(num)
    test_num = num
    divisor = 2
    factors = []

    while test_num > 1
      if (test_num % divisor).zero?
        test_num /= divisor
        factors << divisor
        # retry successful factor
        divisor -= 1
      end
      divisor += 1
    end
    factors
  end
end
